crates/typst-library/src/text/shaping.rs

```diff
@@ -432,9 +432,10 @@ impl<'a> ShapedText<'a> {
     pub fn push_hyphen(&mut self, vt: &Vt) {
         families(self.styles).find_map(|family| {
             let world = vt.world;
-            let font = world
-                .book()
+            let book = world.book();
+            let font = book
                 .select(family.as_str(), self.variant)
+                .or(book.select_fallback(None, self.variant, "-"))
```

Comment:
this will prefer the fallback over the second family in the list. performance-wise, it would also be good to only perform the fallback logic lazily if no match is found.

Reply:
the lazy match is updated. One question though about fallback to the second family: why is that the case? I thought that the select_fallback function should return the best fit.

(Do you mean is it that we perhaps need to feed it a Font `like` to it?)

Reply:
What I meant is the following: The large find_map searches for a font family that has the hyphen in the fallback chain. 

However, this last resort fallback code now already runs when the first family in the chain is not found. As long as select_fallback finds something, the find_map never iterates twice. And if the families iterator would be empty, the fallback code wouldn't run at all.

Reply:
oh I see. Will update.